test/numerical/CMakeLists.txt

```diff
@@ -1,86 +1,82 @@
-target_compile_definitions(TestRNN PRIVATE RTMEMREF_INTERNAL_API)
-target_include_directories(TestRNN
-        PRIVATE
-        ${ONNX_MLIR_SRC_ROOT}/include)
-target_link_libraries(TestRNN
-        rapidcheck
-        MainUtils
-        ExecutionSession
-        OMTensorUtils)
+  add_test(NAME ${test_name} COMMAND ${test_name} WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR})
+  set_tests_properties(${test_name} PROPERTIES LABELS numerical)
 
-add_executable(TestGRU TestGRU.cpp)
-target_compile_definitions(TestGRU PRIVATE RTMEMREF_INTERNAL_API)
-target_include_directories(TestGRU
-        PRIVATE
-        ${ONNX_MLIR_SRC_ROOT}/include)
-target_link_libraries(TestGRU
-        rapidcheck
-        MainUtils
-        ExecutionSession
-        OMTensorUtils)
+  if (WIN32)
+    # On Windows, we need a .def file to specify the export functions from a DLL.
+    # The name of the .def file must match the name of the DLL being created, so
+    # we follow the same naming convention as the tests (e.g. TestConv_main_graph)
+    # (see SHARED_LIB_BASE in each of the test files).
+    configure_file(
+      ${CMAKE_CURRENT_SOURCE_DIR}/numerical.def
+      ${CMAKE_CURRENT_BINARY_DIR}/${test_name}_main_graph.def
+      COPYONLY
+      )
+  endif()
+endfunction()
 
-add_executable(TestLoop TestLoop.cpp)
-target_compile_definitions(TestLoop PRIVATE RTMEMREF_INTERNAL_API)
-target_include_directories(TestLoop
-        PRIVATE
-        ${ONNX_MLIR_SRC_ROOT}/include)
-target_link_libraries(TestLoop
-        rapidcheck
-        MainUtils
-        ExecutionSession
-        OMTensorUtils)
+# All libraries and executables coming from llvm or ONNX MLIR have had their
+# compile flags updated via llvm_update_compile_flags, so we need to do that to
+# rapidcheck as well, so that we can successfully link against it. Otherwise some
+# of the flags for exceptions (among others) are not set correctly.
+llvm_update_compile_flags(rapidcheck)
 
-add_test(NAME OMTestConv COMMAND TestConv)
-add_test(NAME OMTestMatMul2D COMMAND TestMatMul2D)
-add_test(NAME OMTestGemm COMMAND TestGemm)
-add_test(NAME OMTestLoop COMMAND TestLoop)
-add_test(NAME OMTestGRU COMMAND TestGRU)
-add_test(NAME OMTestLSTM COMMAND TestLSTM)
-add_test(NAME OMTestRNN COMMAND TestRNN)
+set(TEST_LINK_LIBS rapidcheck MainUtils ExecutionSession)
+
+add_numerical_unittest(TestConv
+  TestConv.cpp
+  LINK_LIBS PRIVATE ${TEST_LINK_LIBS}
+  )
+
+add_numerical_unittest(TestMatMul2D
+  TestMatMul2D.cpp
+  LINK_LIBS PRIVATE ${TEST_LINK_LIBS}
+  )
+
+add_numerical_unittest(TestGemm
+  TestGemm.cpp
+  LINK_LIBS PRIVATE ${TEST_LINK_LIBS}
+  )
+
+add_numerical_unittest(TestLSTM
+  TestLSTM.cpp
+  LINK_LIBS PRIVATE ${TEST_LINK_LIBS}
+  )
+
+add_numerical_unittest(TestRNN
+  TestRNN.cpp
+  LINK_LIBS PRIVATE ${TEST_LINK_LIBS}
+  )
+
+add_numerical_unittest(TestGRU
+  TestGRU.cpp
+  LINK_LIBS PRIVATE ${TEST_LINK_LIBS}
+  )
+
+add_numerical_unittest(TestLoop
+  TestLoop.cpp
+  LINK_LIBS PRIVATE ${TEST_LINK_LIBS}
+  )
```

Comment:
Function `add_numerical_unittest` is defined with one parameter but called with three arguments?

Reply:
Yes, the rest of the arguments are passed through to add_onnx_mlir_executable. By not formally declaring all the arguments, we don't have to specify what they are and how to parse them and can use ARGN to get the whole collection of arguments past the first and pass them to the function that creates the executable.

Reply:
Why not just write
```
add_onnx_mlir_executable(${test_name} NO_INSTALL ${test_name}.cpp LINK_LIBS PRIVATE ${TEST_LINK_LIBS})
```
and then simply write
```
add_numerical_unittest(TestConv)
add_numerical_unittest(TestMatMul2D)
...
```

Reply:
@gongsu832 : I wanted the usage to look more like add_executable or add_library. If you prefer me to change the usage to just the name of the test, I can do that.

Reply:
No that's fine. Just wondering the rational behind. Perhaps add what you just said as a comment to `add_numerical_unittest`? Thanks.